include/class.dept.php

```diff
@@ -359,6 +373,23 @@ function canAppendSignature() {
         return ($this->getSignature() && $this->isPublic());
     }
 
+    //Check if an agent is eligible for assignment
+    function canAssign(Staff $assignee) {
+        //Primary members only
+        if ($this->assignPrimaryOnly() && !$this->isPrimaryMember($assignee))
+            return false;
+
+        //Extended members only
+        if ($this->assignMembersOnly() && !$this->isMember($assignee))
+            return false;
+
+         //Make sure agent is active & not on vacation
+         if (!$assignee->isActive() || $assignee->onVacation())
```

Comment:
Ouch... looks like you copied and pasted what I had. It was pseudo code.

Please change this like to `if (!$assignee->isAvailable())` 

Might want to also consider that assignee might be a team??